### ISSUE TITLE
overview broken

### ISSUE THREAD

**Issue by caetano-m** [yilozt/rounded-window-corners#14]
I have no direct evidence that this is caused by this extension and only that, but it happens only when the extension is installed and stops when removed. At some point this is what happens to the overview:
![Captura de tela de 2022-08-03 09-05-38](https://user-images.githubusercontent.com/71804353/182709022-d0db6308-463c-4dbc-b787-f587f785c5d8.png)

This is what the log journal shows when I try to show the overview in this state: ```
JS ERROR: Error: Expected an object of type ClutterActorBox for argument 'box' but got type undefined vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:202:33```

As shown in the below screenshot: 
![Captura de tela de 2022-08-03 17-44-50](https://user-images.githubusercontent.com/71804353/182709616-ad1beb27-8678-4b52-b2a3-b977bcbe508a.png)

I'm running Fedora 36 with Gnome 42.3

**Comment by yilozt** [comment]
Weird. Have you try to open overview when only this extensions enabled? If it still break overview, it means that there is a bug about this extensions.

**Comment by oschl-git** [comment]
I've noticed the same issue - while using this extension, after locking the screen, the overview became completely broken. I wanted to contribute something meaningful, so I tried the extension in isolation without any other extension enabled (which didn't produce the bug), and now I can't reproduce it at all anymore. I am not sure why or how, but disabling and re-enabling my other extensions seems to have fixed this. I can however confirm that this bug is related to this extension.

**Comment by caetano-m** [comment]
@oschl-git I suspect it happens when this extension an some other is used at the same time. What extensions are you using? Maybe we can narrow down to the extensions we both are using. I use the below:

- AppIndicator and KStatusNotifierItem Support
- Burn My Windows
- Clipboard History
- Emoji Selector
- Gnome 4x UI Improvements
- GSConnect
- Lock Keys
- Quick Close in Overview
- Status Area Horizontal Spacing
- Steal My Focus 
- Transparent Top Bar

**Comment by caetano-m** [comment]
Ok, I found it! It is the [burn my windows](https://github.com/Schneegans/Burn-My-Windows) extension along with this one!
To replicate the bug:

1. Activate both extensions
2. Lock the screen and unlock it
3. Try to access the overview a couple of times

**Comment by yilozt** [comment]
The reason Burn my windows and this extensions both monkey patch `imports.ui.workspace.Workspace.prototype._addWindowClone`, It cause burn my windows recursion itself when unlock screen if two extension enabled together.

https://github.com/Schneegans/Burn-My-Windows/blob/fbeb40c2d0b84929401a3a24afd39af181c20f7d/extension.js#L84

https://github.com/yilozt/rounded-window-corners/blob/508abeece85ad78dcb91dc8f24365fe309187c19/src/app.ts#L55

Maybe It's relate to the order of extensions enable / disable.

There is my log when I follow the guide of @caetano-m, after unlock it reproduce this issue:

```
[RoundedCordersEffect] Loaded.
[RoundedCornersEffect] DBus Services exported
[RoundedCornersEffect] Enabled
JS ERROR: too much recursion
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:136:51
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
enable/Workspace.prototype._addWindowClone@/home/luo/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/extension.js:137:56
```

I monkey patch `imports.ui.workspace.Workspace.prototype._addWindowClone` is because we need add custom shadow into overview. I will try to find another way to do it.

**Comment by yilozt** [comment]
Should fix in #22

**Comment by oschl-git** [comment]
Awesome, thank you!
